decision-records/0004-data-api.md

```diff
@@ -0,0 +1,113 @@
+
+Currently, the terms annotated by ToS;DR are currently taken from original web documents stored on the ToS;DR server database. Open Terms Archive provides a significantly higher quality source for terms, for example by cleaning up all content that is not related to the terms, combining documents scattered across several pages, or normalising PDF documents.
+
+To allow the transition from historical ToS;DR's crawler and database to the publicly accessible OTA datasets, ToS;DR requires access to the content of the versions through an API. This RFC outlines the proposed solutions.
+
+## Proposed solutions
+
+### Solution A:
+
+#### Base URL
+
+`<collection host>/api/:version`
+
+#### Endpoint
+
+##### `GET /version/:serviceId/:termsType/:date.md`
+
+Return the markdown content of the version valid for the given date
+
+##### Parameters
+
+| Parameter | Type   | Description            |
+| --------- | ------ | ---------------------- |
+| serviceId | URL-encoded string | The ID of the service |
+| termsType | URL-encoded string | The name of terms type |
+| date | URL-encoded ISO date string | The date for which you want the version. This date should not be in the futur |
+
+Note about the date:
+- It is not required that this date match exactly the fetch date of a version. As versions are fetched at periodic interval the version that will be returned will corresond to the version valid for this date range.
+- To have the latest version available, simple use the current date time as parameter.
+
+##### Returns
+
+- A raw markdown content when a version was found for this date
+- A HTTP 404 for a date that is anterior of the first available version
+- A HTTP 400 Bad Request, if the date is in the futur
+
+##### Example
+
+```
+GET /version/Open%20Terms%20Archive/Privacy%20Policy/2023-10-24T14%3A41%3A42.381Z.md
+```
+```markdown
+Privacy Policy
+==============
+
+Last updated: September 26, 2023
+
+This privacy policy explains how, why and under what conditions the Open Terms Archive (OTA) site collects personal information and how it is used. Our privacy policy will change over time. And of course, we also record the changes of [our own documents](https://github.com/OpenTermsArchive/demo-versions/tree/main/Open%20Terms%20Archive) 😉
+
+…
+```
+
+### Solution B:
+
+#### Base URL
+
+`<collection host>/api/:version`
+
+#### Endpoint
+
+##### `GET /version/:serviceId/:termsType/:date`
+
+Return a JSON object containing of the version content valid for the given date along some metadata
+
+##### Parameters
+
+| Parameter | Type   | Description            |
+| --------- | ------ | ---------------------- |
+| serviceId | URL-encoded string | The ID of the service |
+| termsType | URL-encoded string | The name of terms type |
+| date | URL-encoded ISO date string | The date for which you want the version. This date should not be in the futur |
+
+Note about the date:
```

Comment:
```suggestion
Note about the date:
- A full date and time is required, and not a simple date (such as `2023-10-24`), to avoid ambiguities on days where a version changed, and timezone differences between client and server.
```